homeassistant/components/sensor/travel_time.py

```diff
@@ -0,0 +1,114 @@
+"""
+Support for RFXtrx sensors.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.travel_time/
+"""
+from datetime import datetime
+from datetime import timedelta
+import logging
+import voluptuous as vol
+
+from homeassistant.helpers.entity import Entity
+from homeassistant.const import CONF_API_KEY, TEMP_CELSIUS
+from homeassistant.util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['googlemaps==2.4.3']
+
+# Return cached results if last update was less then this time ago
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=5)
+
+CONF_ORIGIN = 'origin'
+CONF_DESTINATION = 'destination'
+CONF_TRAVEL_MODE = 'travel_mode'
+
+PLATFORM_SCHEMA = vol.Schema({
+    vol.Required('platform'): 'travel_time',
+    vol.Required(CONF_API_KEY): vol.Coerce(str),
+    vol.Required(CONF_ORIGIN): vol.Coerce(str),
+    vol.Required(CONF_DESTINATION): vol.Coerce(str),
+    vol.Optional(CONF_TRAVEL_MODE, default='driving'):
+        vol.In(["driving", "walking", "bicycling", "transit"])
+})
+
+
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """Setup the travel time platform."""
+    # pylint: disable=too-many-locals
+
+    is_metric = (hass.config.temperature_unit == TEMP_CELSIUS)
+    api_key = config.get(CONF_API_KEY)
+    origin = config.get(CONF_ORIGIN)
+    destination = config.get(CONF_DESTINATION)
+    travel_mode = config.get(CONF_TRAVEL_MODE)
+
+    sensor = TravelTimeSensor(api_key, origin, destination,
+                              travel_mode, is_metric)
+    add_devices_callback([sensor])
+
+
+class TravelTimeSensor(Entity):
+    """Representation of a tavel time sensor."""
+
+    # pylint: disable=too-many-arguments
+    def __init__(self, api_key, origin, destination, travel_mode, is_metric):
+        """Initialize the sensor."""
+        if is_metric:
+            self._unit = 'metric'
+        else:
+            self._unit = 'imperial'
+        self._origin = origin
+        self._destination = destination
+        self._travel_mode = travel_mode
+        self._name = "Travel time"
+        self._matrix = None
+
+        import googlemaps
+        self._client = googlemaps.Client(api_key, timeout=10)
```

Comment:
Would it be possible to validate the API key in the `setup_platform` method and stop setup if invalid ?